Rework typeclass documentation

aesteve opened the pull request:
This is a work in progress before submitting to Dotty.

Please let me know what you're thinking about it.

anatoliykmetyuk committed:
Release Dotty 0.21.0-RC1

Commit:
trying to rewrite typeclasses-new

odersky committed:
Update kind projector check file

odersky committed:
Update repltest

odersky committed:
Update signature help test

odersky committed:
Implement base variance ops

 + refactoring: account for Bivariance
 + refactpring: collect variance related ops in Variances,
   move Variances to core.

odersky committed:
rename `paramVariance` -> `paramVarianceSign`

odersky committed:
Add infrastructure for setting paramVariances in HKTypeLambdas

This is meant as a better alternative to encode variances in parameter names.

odersky committed:
Parse and print variances according to new scheme

The old encoding using semantic parameter name is still in place. The
new recording inside TypeBounds exists alongside the old one.

odersky committed:
Tasty pickling of variances in TypeBounds

odersky committed:
Abstract over paramVariance instead of paramVarianceSign

odersky committed:
Don't encode variances in names

Encode them instead in the upper bound lambda of a TypeBounds type.

For now, we also encode them in the alias of a Typealias type, but
this will be dropped one we pass to structural lambda variance.

odersky committed:
Make useVariances tunable

Needs to be passed as a parameter since Namer does not construct
a bounds immediately.

odersky committed:
Fix unpickling of type aliases

odersky committed:
Blacklist 3 pickling tests

These can hopefully be revived once the variance changes have completed

odersky committed:
Switch to structural variances for type lambdas

odersky committed:
Drop VariantNames

odersky committed:
Treat type lambdas structurally ...

... unless they are on the right hand side of type bounds or
match aliases or they are type aliases where some variance is
given explicitly with a `+` or `-`.

odersky committed:
Don't treat opaque aliases structurally

odersky committed:
Revert "Blacklist 3 pickling tests"

This reverts commit 7d6e75e1c1c14d05fd11d0d0134595dcddd83eb9.

odersky committed:
Add fixed issue tests

odersky committed:
Update check files

Type lambdas don't print with variances anymore

odersky committed:
Drop adaptHkVariance

odersky committed:
Simplify EtaExpand

odersky committed:
Disallow variances in type lambdas

Since variances are associated conceptually with higher-kinded type variables,
it makes no sense to write them on type lambdas. I believe it's better to disallow
writing variances there because it will only need to variance-bike-shedding otherwise.

odersky committed:
Further simplifications and test fixes

odersky committed:
Improve isRef queries against top types

A refinement type would previously qualify as an `isRef` of
Any, AnyKind, or AnyRef. Often that is not what was intended.
Break out `isAny`, `isAnyKind`, and `isAnyRef` methods for
tests that don't go through refinements.

odersky committed:
Simplify pickling of type aliases

odersky committed:
Don't stop at LazyRefs when computing structural variances

odersky committed:
Improve variance computation and printing

odersky committed:
Fix #8096: Change syntax of single-case try

odersky committed:
Fix #8067: Don't constrain with types of wrong kinds

The example got into a pickle since it tried to constrain a type variable
A[_] with a type (G[_] => G[String]). It should have rejected this outright.
Instead the constraint addition succeeded and the compiler got into a dead-end
with it afterwards.

odersky committed:
Generalize EtaExpand

make it work for al kinds of ParamInfos.

odersky committed:
Refine same-kind criterion

 - Need to take of AnyKind
 - need to also copmare results

odersky committed:
Fix #8129: Refine atoms computation

It makes a difference whether a type appears to the left or to the right
of a <:< comparison. On the left, an abstract type upper bounded by some
atoms can assume to have these atoms. But on the right, that's not the case.

odersky committed:
Fix -Yexplain-types

nicolasstucki committed:
Flatten the tasty.Reflection cake

Using the cake pattern made it easy to split and keep into smaller files.
But this comes at a cost:
 * Extra bridges
 * Not knowing where to find stuff
 * Hard to update
 * Inability to use opaque types and proper extension methods

The downside is that we will have a larger `Reflection.scala` file, but
it will be proportional to the size of `CompilerInterface.scala` which has
been easier to manage than the cake.

Note: `scala.tasty.Reflection` did not have any API changes.

smarter committed:
Upgrade to ASM 7.3.1

Keeping up with scalac: https://github.com/scala/scala/pull/8676

nicolasstucki committed:
Remove `implicit` from tasty.Reflection

nicolasstucki committed:
Use extension methods

anatoliykmetyuk committed:
Merge pull request #8140 from dotty-staging/upgrade/asm-7.3.1

Upgrade to ASM 7.3.1

liufengyun committed:
Merge pull request #8138 from dotty-staging/uncake-tasty-reflection

Flatten the tasty.Reflection cake

anatoliykmetyuk committed:
Add 22nd release article

odersky committed:
Address review comments

odersky committed:
Add comment link to test

philipschwarz commented:
hello - FWIW, related, what I came up with while experimenting: https://www.slideshare.net/pjschwarz/scala-3-by-example-better-semigroup-and-monoid#31

![image](https://user-images.githubusercontent.com/116804/73480126-3cf17f00-4391-11ea-9c03-71462b5ca255.png)

bishabosha committed:
Merge pull request #8121 from dotty-staging/fix-#8096

Fix #8096: Change syntax of single-case try

bishabosha committed:
fix #8096: add regression

nicolasstucki committed:
Fix paths of blogpost images

nicolasstucki committed:
Fix sbt-dotty tasty-inspector-example-project

Commit:
Wrap the last sentence in a whole "summary" section to avoid stray text

anatoliykmetyuk committed:
Merge pull request #8154 from dotty-staging/fix-sbt-dotty-tasty-inspector-example-project

Fix sbt-dotty tasty-inspector-example-project

Commit:
Step2: rewrite the Functor part in more details

nicolasstucki committed:
Apply suggestions from code review

Co-Authored-By: Anatolii Kmetiuk <anatoliykmetyuk@gmail.com>

Commit:
Step3: rewrite the Monad part (apart from the Reader monad)

Commit:
Fix typos

odersky committed:
Fix typos

Commit:
Rephrase assertTransformation simplification

Commit:
Fix typos on the reader monad

nicolasstucki committed:
Merge pull request #8153 from dotty-staging/fix-#8141

Fix paths of blogpost images

Commit:
@bishabosha's note

nicolasstucki committed:
Address review comments

nicolasstucki committed:
Use updated extension syntax

Commit:
attempt at explaining the reader monad

nicolasstucki committed:
Merge pull request #8142 from dotty-staging/remove-implicits-from-tasty-reflection

Remove implicits from tasty reflection

nicolasstucki committed:
Merge pull request #8156 from dotty-staging/use-updated-extension-syntax

Use updated extension syntax

odersky committed:
Remove stray comment line

odersky committed:
Change docs to given/using syntax

aesteve committed:
less concrete and more accurate definition of a Functor

Co-Authored-By: Jamie Thompson <bishbashboshjt@gmail.com>

aesteve committed:
definition of A functor, not THE functor ability

Co-Authored-By: Jamie Thompson <bishbashboshjt@gmail.com>

aesteve committed:
A functor for the type constructor F[_]

Co-Authored-By: Jamie Thompson <bishbashboshjt@gmail.com>

aesteve committed:
better phrasing for abstracting away Config

Co-Authored-By: Jamie Thompson <bishbashboshjt@gmail.com>

aesteve committed:
parameterised type with abstract members => trait

Co-Authored-By: Jamie Thompson <bishbashboshjt@gmail.com>

aesteve committed:
oo polymorphism vs. parametric polymorphism

Co-Authored-By: Jamie Thompson <bishbashboshjt@gmail.com>

aesteve committed:
explaining the difference between OO polymorphism and ad-hoc polymorphism

Co-Authored-By: Jamie Thompson <bishbashboshjt@gmail.com>

aesteve committed:
better phrasing for conclusion

Co-Authored-By: Jamie Thompson <bishbashboshjt@gmail.com>

aesteve committed:
remove the "we don't care" part

Co-Authored-By: Jamie Thompson <bishbashboshjt@gmail.com>

aesteve committed:
using F as a substitution for every type that ca be mapped over

Co-Authored-By: Jamie Thompson <bishbashboshjt@gmail.com>

bishabosha committed:
Add test for tasty tags POLYType METHODtype GIVENMETHODtype and SUPERtype

odersky committed:
Merge pull request #8082 from dotty-staging/change-lambda-variance

Rework variances of higher-kinded types

odersky committed:
Merge pull request #8130 from dotty-staging/fix-#8067

Fix #8067: Don't constrain with types of wrong kinds

nicolasstucki committed:
Fix comments

nicolasstucki committed:
Fix indentation

nicolasstucki committed:
Fix docs indentation

odersky committed:
Fix subsection title

nicolasstucki committed:
Homogenize section commnet tags

odersky committed:
Implement given/using syntax

odersky committed:
Go back to `(given ...` parameters and arguments

This reverts commit 04095bc255351f2764291e3c37a27e0a76eab547.

# Conflicts:
#	compiler/src/dotty/tools/dotc/parsing/Parsers.scala
#	compiler/src/dotty/tools/dotc/parsing/Tokens.scala
#	docs/docs/internals/syntax.md
#	tests/neg-macros/delegate-match-1/Macro_1.scala
#	tests/neg-macros/delegate-match-2/Macro_1.scala
#	tests/neg-macros/delegate-match-3/Macro_1.scala
#	tests/neg-macros/i6432/Macro_1.scala
#	tests/neg-macros/i6432b/Macro_1.scala
#	tests/neg-macros/i6976/Macro_1.scala
#	tests/neg-macros/inline-case-objects/Macro_1.scala
#	tests/neg-macros/inline-macro-staged-interpreter/Macro_1.scala
#	tests/neg-macros/inline-option/Macro_1.scala
#	tests/neg-macros/inline-tuples-1/Macro_1.scala
#	tests/neg-macros/macros-in-same-project-6/Foo.scala
#	tests/neg-macros/quote-error-2/Macro_1.scala
#	tests/neg-macros/quote-error/Macro_1.scala
#	tests/neg-macros/quote-exception/Macro_1.scala
#	tests/neg-macros/quote-whitebox/Macro_1.scala
#	tests/neg-macros/reflect-inline/assert_1.scala
#	tests/neg-macros/tasty-macro-assert-1/quoted_1.scala
#	tests/neg-macros/tasty-macro-assert-2/quoted_1.scala
#	tests/neg-macros/tasty-macro-error/quoted_1.scala
#	tests/neg-macros/tasty-macro-positions/quoted_1.scala
#	tests/neg-macros/tasty-string-interpolator-position-a/Macro_1.scala
#	tests/neg-macros/tasty-string-interpolator-position-b/Macro_1.scala
#	tests/neg/given-eta.scala
#	tests/neg/i7919.scala
#	tests/neg/implicit-params.scala
#	tests/pending/run/tasty-comments/quoted_1.scala
#	tests/pos-custom-args/erased/i7868.scala
#	tests/pos-macros/i6171/Macro_1.scala
#	tests/pos-macros/i6535/Macro_1.scala
#	tests/pos-macros/i6803b/Macro_1.scala
#	tests/pos-macros/i7011/Macros_1.scala
#	tests/pos-macros/quote-nested-object/Macro_1.scala
#	tests/pos-macros/quote-whitebox-2/Macro_1.scala
#	tests/pos-macros/tasty-constant-type/Macro_1.scala
#	tests/pos-macros/treemap-unapply/Macro.scala
#	tests/pos/i7204.scala
#	tests/pos/i7532.scala
#	tests/run-custom-args/Yretain-trees/tasty-definitions-2/Macro_1.scala
#	tests/run-custom-args/Yretain-trees/tasty-definitions-3/Macro_1.scala
#	tests/run-custom-args/Yretain-trees/tasty-extractors-owners/quoted_1.scala
#	tests/run-custom-args/Yretain-trees/tasty-load-tree-1/quoted_1.scala
#	tests/run-custom-args/Yretain-trees/tasty-load-tree-2/quoted_1.scala
#	tests/run-custom-args/run-macros-erased/reflect-isFunctionType/macro_1.scala
#	tests/run-macros/f-interpolation-1/FQuote_1.scala
#	tests/run-macros/gestalt-type-toolbox-reflect/Macro_1.scala
#	tests/run-macros/i4735/Macro_1.scala
#	tests/run-macros/i4803/Macro_1.scala
#	tests/run-macros/i4803b/Macro_1.scala
#	tests/run-macros/i4803c/Macro_1.scala
#	tests/run-macros/i5119/Macro_1.scala
#	tests/run-macros/i5119b/Macro_1.scala
#	tests/run-macros/i5188a/Macro_1.scala
#	tests/run-macros/i5533/Macro_1.scala
#	tests/run-macros/i5533b/Macro_1.scala
#	tests/run-macros/i5536/Macro_1.scala
#	tests/run-macros/i5629/Macro_1.scala
#	tests/run-macros/i5715/Macro_1.scala
#	tests/run-macros/i5941/macro_1.scala
#	tests/run-macros/i6171/Macro_1.scala
#	tests/run-macros/i6270/Macro_1.scala
#	tests/run-macros/i6518/Macro_1.scala
#	tests/run-macros/i6679/Macro_1.scala
#	tests/run-macros/i6765-c/Macro_1.scala
#	tests/run-macros/i6765/Macro_1.scala
#	tests/run-macros/i6988/FirstArg_1.scala
#	tests/run-macros/i7898/Macro_1.scala
#	tests/run-macros/i7964/Macro_1.scala
#	tests/run-macros/inferred-repeated-result/test_1.scala
#	tests/run-macros/inline-case-objects/Macro_1.scala
#	tests/run-macros/inline-macro-staged-interpreter/Macro_1.scala
#	tests/run-macros/inline-option/Macro_1.scala
#	tests/run-macros/inline-tuples-1/Macro_1.scala
#	tests/run-macros/inline-tuples-2/Macro_1.scala
#	tests/run-macros/inline-varargs-1/Macro_1.scala
#	tests/run-macros/quote-and-splice/Macros_1.scala
#	tests/run-macros/quote-inline-function/quoted_1.scala
#	tests/run-macros/quote-matcher-power/Macro_1.scala
#	tests/run-macros/quote-matcher-runtime/quoted_1.scala
#	tests/run-macros/quote-simple-macro/quoted_1.scala
#	tests/run-macros/quote-toExprOfTuple/Macro_1.scala
#	tests/run-macros/quote-type-matcher/quoted_1.scala
#	tests/run-macros/quote-whitebox/Macro_1.scala
#	tests/run-macros/quoted-expr-block/quoted_1.scala
#	tests/run-macros/quoted-matching-docs/Macro_1.scala
#	tests/run-macros/refined-selectable-macro/Macro_1.scala
#	tests/run-macros/reflect-dsl/assert_1.scala
#	tests/run-macros/reflect-inline/assert_1.scala
#	tests/run-macros/reflect-lambda/assert_1.scala
#	tests/run-macros/reflect-pos-fun/assert_1.scala
#	tests/run-macros/reflect-select-constructor/assert_1.scala
#	tests/run-macros/reflect-select-copy-2/assert_1.scala
#	tests/run-macros/reflect-select-copy/assert_1.scala
#	tests/run-macros/reflect-select-symbol-constructor/assert_1.scala
#	tests/run-macros/reflect-select-value-class/assert_1.scala
#	tests/run-macros/reflect-typeChecks/assert_1.scala
#	tests/run-macros/requiredSymbols/Macro_1.scala
#	tests/run-macros/tasty-argument-tree-1/quoted_1.scala
#	tests/run-macros/tasty-custom-show/quoted_1.scala
#	tests/run-macros/tasty-dealias/quoted_1.scala
#	tests/run-macros/tasty-definitions-1/quoted_1.scala
#	tests/run-macros/tasty-eval/quoted_1.scala
#	tests/run-macros/tasty-extractors-1/quoted_1.scala
#	tests/run-macros/tasty-extractors-2/quoted_1.scala
#	tests/run-macros/tasty-extractors-3/quoted_1.scala
#	tests/run-macros/tasty-extractors-types/quoted_1.scala
#	tests/run-macros/tasty-getfile/Macro_1.scala
#	tests/run-macros/tasty-interpolation-1/Macro.scala
#	tests/run-macros/tasty-linenumber-2/quoted_1.scala
#	tests/run-macros/tasty-linenumber/quoted_1.scala
#	tests/run-macros/tasty-location/quoted_1.scala
#	tests/run-macros/tasty-macro-assert/quoted_1.scala
#	tests/run-macros/tasty-macro-const/quoted_1.scala
#	tests/run-macros/tasty-macro-positions/quoted_1.scala
#	tests/run-macros/tasty-original-source/Macros_1.scala
#	tests/run-macros/tasty-seal-method/quoted_1.scala
#	tests/run-macros/tasty-simplified/quoted_1.scala
#	tests/run-macros/tasty-subtyping/quoted_1.scala
#	tests/run-macros/tasty-tree-map/quoted_1.scala
#	tests/run-macros/tasty-typeof/Macro_1.scala
#	tests/run-macros/tasty-unsafe-let/quoted_1.scala
#	tests/run-macros/type-show/Macro_1.scala
#	tests/run-macros/xml-interpolation-1/XmlQuote_1.scala
#	tests/run-macros/xml-interpolation-2/XmlQuote_1.scala
#	tests/run-macros/xml-interpolation-3/XmlQuote_1.scala
#	tests/run-with-compiler/i6201/macro_1.scala
#	tests/run/given-eta.scala

odersky committed:
Fix parsing conditional givens with `using`

odersky committed:
Partial comversion to using clauses

Convert all files that had a conflict in the last commit, which reverted
`with` clauses to `given` parameters.

bishabosha committed:
Merge pull request #8160 from dotty-staging/add-regression-tasty-tags-obscure

Add pos test to generate rare tasty tags

nicolasstucki committed:
Merge pull request #8161 from dotty-staging/fix-comments-and-identation

Fix comments and indentation

nicolasstucki committed:
Fix nightly build for TASTy Inspector

nicolasstucki committed:
Merge pull request #8164 from dotty-staging/fix-nightly-tasty-inspector

Fix nightly build for TASTy Inspector

odersky committed:
Change remaining uses of given to using

odersky committed:
Merge pull request #8149 from dotty-staging/issue/8096/add-regression

fix #8096: add regression

odersky committed:
Remove vestiges of `with` in printing and patching code

odersky committed:
Fix #8151: Make `setGivenApply` more flexible.

odersky committed:
Fix #8150: Improve screening of legal refinements

odersky committed:
Improve error message

valencik committed:
Add missing bracket

valencik committed:
Fix typo in noAutoTupling

valencik committed:
Fix typo in "rely on an internal..."

odersky committed:
Merge pull request #8167 from valencik/scala-shadowing-typo

Fix typos in language.scala

odersky committed:
Merge pull request #8162 from dotty-staging/change-given-using

Change to given/using syntax

odersky committed:
Some refactorings involvng TempInfos

nicolasstucki committed:
Stop documentation on merge

We will only have documentation generated on each RC and nightly release

odersky committed:
Instantiate tempInfo later

gzoller committed:
Addressed issue 8163 to add opaque type alias information to Reflection API

odersky committed:
Refactorings in preparation of opaques change

odersky committed:
Fix CyclicError error message if denotation does not exist

odersky committed:
Fix printing of HKTypeLambdas

anatoliykmetyuk committed:
Merge branch '0.21.x'

anatoliykmetyuk committed:
Set baseVersion to 0.23.0

anatoliykmetyuk committed:
Merge pull request #8171 from dotty-staging/stop-documentation-on-merge

Stop documentation on merge

gzoller committed:
Reflecting requested changes

dwijnand committed:
docs: Fix a typo in motivation.md

dwijnand committed:
Fixup the description of "Using Clauses"

nicolasstucki committed:
Add test

Implementation of quoted pattern matching formalization.

Additions:
 * `T-Abs`, `T-Pat-Var` and `T-Pat-Abs` require a context that does not already define their name
 * `Fix` and required renaming

odersky committed:
Better support for LazyRefs in & and |

liufengyun committed:
Fix #8178: add test

Cannot reproduce on master

nicolasstucki committed:
Merge pull request #8180 from dwijnand/patch-2

docs: Fix a typo in motivation.md

nicolasstucki committed:
Merge pull request #8184 from dotty-staging/fix-8178

Fix #8178: add test

nicolasstucki committed:
Fix #8186: Check trait secondary constructors

nicolasstucki committed:
Use extensions for show* in Reflection

Workaround #8188 by renaming on of the `show` to `showWith`

odersky committed:
Update docs to 0.22RC

nicolasstucki committed:
Move ConstantShowDeco into ConstantOps

nicolasstucki committed:
Move TreeShowDeco into TreeOps

nicolasstucki committed:
Move TypeOrBoundsShowDeco to TypeOrBoundsOps

nicolasstucki committed:
Move SymbolShowDeco into SymbolOps

nicolasstucki committed:
Move FlagsShowDeco inot FlagsOps

nicolasstucki committed:
Add missing contexts

nicolasstucki committed:
Do not check java trait secondary constructors

See tests/pos/i851.java

odersky committed:
Switch to new scheme for handling opaques in Tasty reader

nicolasstucki committed:
Rename *ImplicitFunctionN to *ContextFunctionN

nicolasstucki committed:
Merge pull request #8190 from dotty-staging/use-extension-in-reflection

Use extensions for show* in Reflection

nicolasstucki committed:
Merge pull request #8191 from dotty-staging/add-missing-contexts

Add missing contexts and remove unnecessary given arguments

nicolasstucki committed:
Merge pull request #8187 from dotty-staging/fix-#8186

Fix #8186: Check trait secondary constructors

anatoliykmetyuk committed:
Merge pull request #8189 from dotty-staging/update-dcos

Update docs to 0.22RC

nicolasstucki committed:
Extract Tree utils from Reflection

* Allows TreeMap, TreeTraverser and TreeAccumulator to be extended in non local classes
* Removes logic that does not belong in Reflection

nicolasstucki committed:
Merge pull request #8176 from gzoller/i8163_opaque_types

Addressed issue 8163 to add opaque alias info to Reflection API

OlivierBlanvillain committed:
Merge pull request #8137 from dotty-staging/fix-#8129

Fix #8129: Refine atoms computation

odersky committed:
Fix #8152: Fix type parameters for opaque bounds

odersky committed:
Cleanups

odersky committed:
Merge pull request #8192 from dotty-staging/rename-implicit-functions

Rename *ImplicitFunctionN to *ContextFunctionN

odersky committed:
More cleanups

nicolasstucki committed:
Fix #8198: Handle trees with TypedSplice

odersky committed:
Fix #8175: Explain transparency of toplevel opaque types

odersky committed:
Fix #8181: Fix isExprIntro predicate to exclude `extension on`

liufengyun committed:
Merge pull request #8200 from dotty-staging/fix-#8198

Fix #8198: Handle trees with TypedSplice

nicolasstucki committed:
Add Reflection constructors for SuperType, RecursiveType and RecursiveThis

nicolasstucki committed:
Fix #7142: Detect escaped variables across macro expansion

anatoliykmetyuk committed:
Upgrade Dotty Reference to 0.22.0-RC1

nicolasstucki committed:
Preextend TreeAccumulator, TreeTraverser and TreeMap in Reflection

odersky committed:
Merge pull request #8202 from dotty-staging/fix-#8181

Fix #8181: Fix isExprIntro predicate to exclude `extension on`

anatoliykmetyuk committed:
Merge pull request #8207 from dotty-staging/reference-update

Upgrade Dotty Reference to 0.22.0-RC1

nicolasstucki committed:
Add tests to scoped quoted Expr prototype

nicolasstucki committed:
Merge pull request #8204 from dotty-staging/add-missing-type-constructors

Add Reflection constructors for SuperType, RecursiveType and RecursiveThis

nicolasstucki committed:
Merge pull request #8193 from dotty-staging/extract-tree-utils-from-reflection

Extract Tree utils from Reflection

odersky committed:
Document usage of OPAQUE tag for classes containing opaque aliases

odersky committed:
Generalize TypeBoundsTree to bounded opaque aliases

The RHS of an opaque type can consist of three parts:
```
  >: loBound <: hiBound = alias
```
This is encoded sp far in WithBounds annotation, but that mechanism means
hidden complexity for the Tasty standard. We now encode the alias directly,
as an optional third argument to a TypeBoundsTree structure.

nicolasstucki committed:
Use new given synatax

nicolasstucki committed:
Use new given syntax

And add missing Context to Term_etaExpand

nicolasstucki committed:
Update given syntax

nicolasstucki committed:
Update given syntax

nicolasstucki committed:
Merge pull request #8205 from dotty-staging/fix-#7142-2

Fix #7142: Detect escaped variables across macro expansion

nicolasstucki committed:
Update given syntax

nicolasstucki committed:
Update docs

liufengyun committed:
Merge pull request #8209 from dotty-staging/add-tests-to-scoped-quoted-expr

Add tests to scoped quoted Expr prototype

liufengyun committed:
Merge pull request #8210 from dotty-staging/update-reflection-given-syntax

Update reflection given syntax

nicolasstucki committed:
Replace given -> using

anatoliykmetyuk committed:
Merge pull request #8145 from dotty-staging/22-release-article

Add the 0.22.0-RC1 release article

odersky committed:
Drop WithBounds annotation

Use new three-element TypeBoundsTree to represent bounded opaque aliases instead.

odersky committed:
Bump Tasty version

robstoll committed:
doc(metaprogramming): typo, add missing `of`

robstoll committed:
doc(metaprogramming): fix opening `

odersky committed:
Address review suggestions

robstoll committed:
doc(metaprogramming): `serialize to` instead of in

robstoll committed:
doc(metaprogramming): stored in

nicolasstucki committed:
Merge pull request #8211 from dotty-staging/update-quoted-given-syntax

Update scala.quoted given syntax

odersky committed:
Merge pull request #8170 from dotty-staging/fix-#8152

Fix #8152: Rearchitect some parts of opaque types handling

nicolasstucki committed:
Merge pull request #8213 from robstoll/patch-16

doc(metaprogramming): typo, add missing `of`

som-snytt committed:
ParseResult ctx gets a compilation unit

Fix #7934

nicolasstucki committed:
Merge pull request #8221 from som-snytt/issue/7934

ParseResult ctx gets a compilation unit

nicolasstucki committed:
Add reference to template TASTy Inspector project

nicolasstucki committed:
Add reference to Dotty Staging template project

nicolasstucki committed:
Fix #8218: Add missing TASTy Inspector dependency in dist/pack

anatoliykmetyuk committed:
Merge pull request #8222 from lampepfl/nicolasstucki-patch-1

Add reference to TASTy Inspector template project

anatoliykmetyuk committed:
Merge pull request #8223 from lampepfl/nicolasstucki-patch-2

Add reference to Dotty Staging template project

nicolasstucki committed:
Merge pull request #8224 from dotty-staging/fix-#8218

Fix #8218: Add missing TASTy Inspector dependency in dist/pack

liufengyun committed:
Merge pull request #8185 from dotty-staging/quoted-semantics

Add test

odersky committed:
Fix #8219: Use translucent supertype to decide AnyKindedness

odersky committed:
Merge pull request #8212 from dotty-staging/drop-withbounds

Generalize TypeBoundsTree to bounded opaque aliases

odersky committed:
Merge pull request #8166 from dotty-staging/fix-#8150

Fix #8150: Improve screening of legal refinements

odersky committed:
Merge pull request #8201 from dotty-staging/fix-#8175

Fix #8175: Explain transparency of toplevel opaque types

odersky committed:
Merge pull request #8165 from dotty-staging/fix-#8151

Fix #8151: Make `setGivenApply` more flexible.

smarter committed:
Merge pull request #8226 from dotty-staging/fix-#8219

Fix #8219: Use translucent supertype to decide AnyKindedness

som-snytt committed:
Fix #8214 Don't consume whitespace after XML

Forward port of https://github.com/scala/scala/pull/4185

nicolasstucki committed:
Update tasty-inspect.md

sjrd committed:
Upgrade to Scala.js 1.0.0.

Contrary to 1.0.0-RC2, Scala.js 1.0.0 respects custom repositories
for resolving artifacts, so regenerating the Docket images is not
necessary anymore.

nicolasstucki committed:
Update tasty-inspect.md

tgodzik committed:
Remove unneccassary cast to URLClassLoader, which breaks on JDK 11

smarter committed:
Merge pull request #8230 from tgodzik/fix-bloop

Remove unneccassary cast to URLClassLoader, which breaks on JDK 11

robstoll committed:
doc(inline): improve log.then description

robstoll committed:
doc(inline): improve recursive example

anatoliykmetyuk committed:
Merge pull request #8227 from lampepfl/nicolasstucki-patch-1

Update tasty-inspect.md

odersky committed:
Merge pull request #8228 from som-snytt/issue/8214

Fix #8214 Don't consume whitespace after XML

nicolasstucki committed:
Update message for erased function types

nicolasstucki committed:
Update to new extension syntax

nicolasstucki committed:
Workaround #8241

nicolasstucki committed:
Update to `given ..: T` to `given .. as T` syntax

nicolasstucki committed:
Update to new extension syntax

smarter committed:
Merge pull request #8229 from dotty-staging/scalajs-1.0.0

Upgrade to Scala.js 1.0.0.

ausmarton committed:
Port Desugar error messages to the new scheme

nicolasstucki committed:
Explicitly provide the given QuoteContext of the splice

 * Avoids leaking implementation details of scala.internal.quoted.CompileTime.exprSplice,
   such as exprSplice taking a ?=> function argument
 * Provide meaningful names for QuoteContext synthesized by within `${ ... }`

nicolasstucki committed:
Add missing assertion for macros

anatoliykmetyuk committed:
Merge pull request #8246 from dotty-staging/add-missing-assertion

Add missing assertion for macros

robstoll committed:
doc(inline specialisation): typos and improvements

robstoll committed:
doc(inline constValue): typo constvalue and toInt

anatoliykmetyuk committed:
Merge pull request #8243 from ausmarton/ticket/1589

Port Desugar error messages to the new scheme

nicolasstucki committed:
Merge pull request #8247 from robstoll/patch-20

doc(inline specialisation): typos and improvements

nicolasstucki committed:
Merge pull request #8248 from robstoll/patch-21

doc(inline constValue): typo constvalue and toInt

nicolasstucki committed:
Merge pull request #8234 from robstoll/patch-17

doc(inline): improve log.then description

odersky committed:
Fix #8241: Allow end markers for extensions

nicolasstucki committed:
Merge pull request #8252 from dotty-staging/fix-#8241

Fix #8241: Allow end markers for extensions

robstoll committed:
doc(inline): typeo usualy instead of usually

Co-Authored-By: Nicolas Stucki <nicolas.stucki@gmail.com>

robstoll committed:
doc(inline): show inline can result in larger code

valencik committed:
Fix #5955: Remove typelevel relationship doc

nicolasstucki committed:
Merge pull request #8254 from valencik/remove-typelevel-doc

Fix #5955: Remove typelevel relationship doc

nicolasstucki committed:
Merge pull request #8235 from robstoll/patch-19

doc(inline): improve recursive example

nicolasstucki committed:
Update inline.md

Make example easier to read

liufengyun committed:
Merge pull request #8259 from lampepfl/nicolasstucki-patch-1

Update inline.md

liufengyun committed:
Merge pull request #8242 from dotty-staging/update-to-given-as-syntax

Update to `given ..: T` to `given .. as T` syntax

liufengyun committed:
Merge pull request #8240 from dotty-staging/update-to-new-extension-syntax

Update to new extension syntax

liufengyun committed:
Merge pull request #8239 from dotty-staging/update-erased-function-parse-error

Update message for erased function types

nicolasstucki committed:
Apply suggestions from code review

Co-Authored-By: Fengyun Liu <liu@fengy.me>

nicolasstucki committed:
Merge pull request #8244 from dotty-staging/handle-quote-context-splice

Explicitly provide the given QuoteContext of the splice

Commit:
Merge remote-tracking branch 'upstream/master' into doc/rework-typeclasses-new

# Conflicts:
#	docs/docs/reference/contextual/typeclasses-new.md

Commit:
Merge remote-tracking branch 'origin/doc/rework-typeclasses-new' into doc/rework-typeclasses-new

Commit:
Trying to add an easy-to-grasp definition of type classes

Commit:
Proper 0.23 syntax

Commit:
Proper 0.23 syntax

Commit:
typo

Commit:
no longer `given as`

Commit:
typeclasses-new.md is now typeclasses.md

Commit:
Adapt to 0.23 latest

Commit:
Use new type wildcard syntax
